### PR TITLE
core: test Modules::deps and handle error cases better

### PR DESCRIPTION
**Pull request by ry**
cc @fewf

### CODE REVIEW

File: cli/main.rs [hunk 1]
```diff
@@ -108,12 +108,19 @@ pub fn print_file_info(worker: &Worker, url: &str) {
...
+    if let Some(ref depsdeps) = deps.deps {
+      for d in depsdeps {
+        println!("{}", d);
+      }
     }
+  } else {
+    println!(
+      "{}{}",
+      ansi::bold("deps: ".to_string()),
+      "cannot retrieve full dependency graph"
```
Comment (piscisaureus) [inline cli/main.rs]:
```
literal with an empty format string

note: #[warn(clippy::print_literal)] on by default
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#print_literal
```

File: core/modules.rs [hunk 2]
```diff
@@ -355,12 +364,25 @@ impl Deps {
 
           Self::helper(seen, new_prefix, new_is_last, modules, dep_name)
         }).collect();
-      Deps {
+
+      // If any of the children are missing, return None.
```
Comment (piscisaureus) [inline core/modules.rs]: Some notes on this section (up to line 378):

* I get the clippy warning: `this block may be rewritten with the `?` operator`. Not sure what it has in mind exactly though.
* The whole thing can be written shorter and more efficiently by relying on the fact that the `collect()` method can turn an iterator over `Option<T>` into an `Option<Vec<T>` (this also works for `Result` and for different types of collections). Replace it with this: `let deps = maybe_deps.into_iter().collect::<Option<_>>()?;`

File: core/modules.rs [hunk 3]
```diff
@@ -321,7 +325,7 @@ pub struct Deps {
 }
 
 impl Deps {
-  pub fn new(modules: &Modules, module_name: &str) -> Deps {
+  fn new(modules: &Modules, module_name: &str) -> Option<Deps> {
```
Comment (piscisaureus) [inline core/modules.rs]: If this function returns an Option<Deps> then I think the field Deps::deps shouldn't be an Option any more.

File: core/modules.rs [hunk 4]
```diff
@@ -332,19 +336,24 @@ impl Deps {
...
-      Deps {
+      Some(Deps {
         name: name.to_string(),
         prefix,
         deps: None,
         is_last,
-      }
+      })
     } else {
+      let maybe_children = modules.get_children2(name);
```
Comment (piscisaureus) [inline core/modules.rs]:
```
this block may be rewritten with the `?` operator

note: #[warn(clippy::question_mark)] on by default
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#question_mark
help: replace_it_with: `maybe_children?;`
```

File: core/modules.rs [hunk 5]
```diff
@@ -332,19 +336,24 @@ impl Deps {
...
+        return None;
+      }
+      let children = maybe_children.unwrap();
+
       seen.insert(name.to_string());
-      let children = modules.get_children2(name).unwrap();
-      let child_count = children.iter().count();
-      let deps = children
+      let child_count = children.len();
+      let mut maybe_deps: Vec<Option<Deps>> = children
```
Comment (piscisaureus) [inline core/modules.rs]:
```
variable does not need to be mutable

note: #[warn(unused_mut)] on by default
help: remove this `mut`: ``
```

**Comment by piscisaureus** [comment]
Suggestions for improvement: https://github.com/denoland/deno/pull/2141/commits/27021f3ea93876a642f2c862d586e2ff6049e601

File: core/modules.rs [hunk 6]
```diff
@@ -354,13 +358,16 @@ impl Deps {
           new_prefix.push(' ');
```
Comment (piscisaureus) [inline core/modules.rs]: The construction of this string would be more readable using `format!()`

**Comment by ry** [comment]
I'm getting link errors locally on this branch - apparently not present in travis
```

Undefined symbols for architecture x86_64:
  "_$LT$$RF$T$u20$as$u20$core..fmt..Debug$GT$::fmt::hd903d53c6ac05420", referenced from:
      l_anon.e0f25427fbcc27a14603ac81d9b78f18.2 in libdeno.rlib(deno.deno.3iaglfa3-cgu.9.rcgu.o)
  "_$LT$$u5b$T$u5d$$u20$as$u20$core..fmt..Debug$GT$::fmt::h6647516962e1187f", referenced from:
      _$LT$alloc..vec..Vec$LT$T$GT$$u20$as$u20$core..fmt..Debug$GT$::fmt::h40987b2ab090dd94 in libdeno.rlib(deno.deno.3iaglfa3-cgu.2.rcgu.o)
  "core::ops::function::impls::_$LT$impl$u20$core..ops..function..FnOnce$LT$A$GT$$u20$for$u20$$RF$mut$u20$F$GT$::call_once::hb43d0dd97edf8624", referenced from:
      _$LT$core..option..Option$LT$T$GT$$GT$::map::he5b8c8863d129f7e in libdeno.rlib(deno.deno.3iaglfa3-cgu.8.rcgu.o)
  "core::ptr::_$LT$impl$u20$$BP$mut$u20$T$GT$::is_null::hc835da8d89975560", referenced from:
      _$LT$alloc..vec..Vec$LT$T$GT$$u20$as$u20$core..ops..deref..Deref$GT$::deref::h760ac621251a2960 in libdeno.rlib(deno.deno.3iaglfa3-cgu.2.rcgu.o)
      _$LT$alloc..vec..Vec$LT$T$GT$$u20$as$u20$core..ops..deref..DerefMut$GT$::deref_mut::h84d5e31ef02e4755 in libdeno.rlib(deno.deno.3iaglfa3-cgu.2.rcgu.o)
  "core::ptr::_$LT$impl$u20$$BP$const$u20$T$GT$::add::h8b140fcd069e22d3", referenced from:
      _$LT$std..collections..hash..table..RawBucket$LT$K$C$$u20$V$GT$$GT$::pair::h5dd26ed0844bbfc9 in libdeno.rlib(deno.deno.3iaglfa3-cgu.1.rcgu.o)
  "core::iter::iterator::Iterator::map::h086a4dbef87d9792", referenced from:
      deno::flags::v8_set_flags::h9b87a6526c2e0ac0 in libdeno.rlib(deno.deno.3iaglfa3-cgu.9.rcgu.o)
  "core::iter::iterator::Iterator::map::h321f9c582823bbfa", referenced from:
      deno::flags::v8_set_flags::h9b87a6526c2e0ac0 in libdeno.rlib(deno.deno.3iaglfa3-cgu.9.rcgu.o)
```

**Comment by ry** [comment]
The above link errors were caused by building with 1.33.0 - it works with 1.34.0 ...

**Pull request merged**